datafusion/src/physical_plan/expressions/min_max.rs

```diff
@@ -72,15 +81,15 @@ impl AggregateExpr for Max {
     fn field(&self) -> Result<Field> {
         Ok(Field::new(
             &self.name,
-            self.data_type.clone(),
+            min_max_aggregate_data_type(self.data_type.clone()),
```

Comment:
since `min_max_aggregate_data_type` is already performing clone internally, perhaps we can just pass by reference here?

Reply:
I think it might be clearer code if this datatype transformation was done as part of the construction of `Min` and `Max` -- that way `self.data_type` would contain the correct output type for the aggregate, and future code would not have to remember to translate `self.data_type`

Perhaps  something like


```
impl Max {
    /// Create a new MAX aggregate function
    pub fn new(
        expr: Arc<dyn PhysicalExpr>,
        name: impl Into<String>,
        data_type: DataType,
    ) -> Self {
        Self {
            name: name.into(),
            expr,
            data_type: min_max_aggregate_data_type(data_type),
            nullable: true,
        }
    }
}
```